src/components/workProgress/WorkProgressInput.tsx

```diff
@@ -0,0 +1,236 @@
+  }
+  return queryInput;
+};
+
+export default function WorkProgressInput({
+  initialValue,
+  onSubmitAction,
+}: {
+  initialValue: WorkProgressQueryInput;
+  onSubmitAction: (submitData: WorkProgressQueryInput) => void;
+}) {
+  //Initialize form data
+  const defaultInitialValues: WorkProgressInputData = {
+    types: Object.values(WorkProgressInputTypeField),
+    values: [],
+    selectedType: WorkProgressInputTypeField.WorkNumber,
+    selectedValue: "",
+    ...mergeFieldTypes(initialValue),
+  };
+  const [current, send] = useMachine(
+    createWorkProgressInputMachine({
+      workProgressInput: defaultInitialValues,
+    })
+  );
+
+  const {
+    workProgressInput: { types, values, selectedType, selectedValue },
+    serverError,
+  } = current.context;
+
+  /**
+   * Form validation schema
+   */
+  const validationSchema: Yup.ObjectSchema = Yup.object().shape({
+    type: Yup.string().ensure(),
+    value: Yup.string().ensure(),
+  });
+
+  const onFormSubmit = () => {
+    onSubmitAction(formatFieldData(current.context.workProgressInput));
+  };
+
+  //Send Events to State machine
+  const sendEvents = (eventType: string, value: string) => {
+    if (eventType === "VALUE_SELECTION") {
+      send({ type: "VALUE_SELECTION", value: value });
+      return;
+    }
+    switch (eventType) {
+      case WorkProgressInputTypeField.WorkNumber: {
+        send({ type: "WORK_NUMBER_SELECTION" });
+        break;
+      }
+      case WorkProgressInputTypeField.WorkType: {
+        send({ type: "WORK_TYPE_SELECTION" });
+        break;
+      }
+      case WorkProgressInputTypeField.Status: {
+        send({ type: "STATUS_SELECTION" });
+        break;
+      }
+    }
+  };
+
+  return (
+    <div className="mx-auto max-w-screen-xl mt-2 my-6 border border-gray-200 bg-gray-100 p-6 rounded-md space-y-4">
+      <Heading level={3} showBorder={false}>
+        Search
+      </Heading>
+
+      <Formik
+        initialValues={defaultInitialValues}
+        validationSchema={validationSchema}
+        validateOnChange={false}
+        validateOnBlur={false}
+        validateOnMount={false}
+        onSubmit={onFormSubmit}
+      >
+        {({ errors, isValid, setFieldValue }) => (
+          <Form>
+            {!isValid && (
+              <Warning className={"mb-5"} message={"Validation Error"}>
+                {Object.values(errors)}
+              </Warning>
+            )}
+            {serverError && (
+              <Warning message="Search Error" error={serverError} />
+            )}
+            <div className="space-y-2 md:px-10 md:space-y-0 md:flex md:flex-row md:justify-center md:items-center md:gap-4">
+              <div className="md:flex-grow">
+                <FormikSelect
+                  label=""
+                  name="type"
+                  data-testid={"type"}
+                  value={selectedType}
+                  onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+                    setFieldValue(
```

Comment:
I'm not quite sure the purpose of the `setFieldValue` here. Isn't `"searchType"` always going to be `undefined`?